### PR TITLE
fix(Resilience4j): upgrade resilience4j version + fix suspend function handling

### PR DESCRIPTION
**Pull request by gal-yardeni**
- upgrade resilience4j version from 1.0.0 to 1.5.0 to support kotlin suspended functions
- fixes https://github.com/spinnaker/keel/issues/1189

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -113,7 +113,9 @@ subprojects {
         "com.squareup.okhttp3:okhttp-urlconnection:$okHttpVersion",
         "com.squareup.okhttp3:okhttp-sse:$okHttpVersion",
         "com.squareup.okhttp3:mockwebserver:$okHttpVersion",
-        "com.squareup.okhttp3:logging-interceptor:$okHttpVersion")
+        "com.squareup.okhttp3:logging-interceptor:$okHttpVersion",
+        "io.github.resilience4j:resilience4j-kotlin:1.5.0",
```
Comment (gal-yardeni) [inline build.gradle.kts:117]: without it here, the version is not forced to be the newer version

File: keel-igor/keel-igor.gradle.kts [hunk 2]
```diff
@@ -9,6 +9,8 @@ dependencies {
   implementation("org.springframework.boot:spring-boot-autoconfigure")
   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core")
 
+  implementation ("io.github.resilience4j:resilience4j-kotlin")
```
Comment (gal-yardeni) [inline keel-igor/keel-igor.gradle.kts:12]: it's not recognizing `io.github.resilience4j.kotlin.retry.decorateSuspendFunction` in `ArtifactMetadataService` without adding this here.

**Review by lorin — APPROVED** [review]
LGTM!

File: keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt [hunk 3]
```diff
@@ -99,13 +97,14 @@ class ArtifactMetadataService(
...
-    ) {
-      runBlocking {
-        buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
-      }
-    }).get()
+
+    //decorate the suspended function and execute it
+    return retry.decorateSuspendFunction {
+      buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
+    } ()
```
Comment (luispollo) [inline keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt:104]: I think this variant is a little more readable:
```suggestion
    return retry.executeSuspendFunction {
      buildService.getArtifactMetadata(commitId = commitId, buildNumber = buildNumber)
    }
```

**Review by luispollo — APPROVED** [review]
Just a minor suggestion, otherwise LGTM.

File: keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt (hunk 3 above)
Comment (gal-yardeni) [reply keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt]: good call!

**Pull request merged**